test/integration/main.test.js

```diff
@@ -377,3 +377,216 @@ test('createRemoteBundle(sources) - other status codes', async () => {
     );
     await closeServer(server);
 });
+
+test('publishAssets(tag, files, options) - 400 error', async () => {
+    expect.assertions(1);
+    const { server, port } = await createTestServer([
+        {
+            verb: 'post',
+            path: '/publish-assets',
+            cb(req, res) {
+                res.status(400).send({ message: 'Bad request' });
+            },
+        },
+    ]);
+    const client = new Client({ buildServerUri: `http://127.0.0.1:${port}` });
+
+    const result = client.publishAssets('podlet1', ['first.js', 'second.js']);
+
+    await expect(result).rejects.toEqual(new Error('Bad request'));
+    await closeServer(server);
+});
+
+test('publishAssets(tag, files, options) - 500 error', async () => {
+    expect.assertions(1);
+    const { server, port } = await createTestServer([
+        {
+            verb: 'post',
+            path: '/publish-assets',
+            cb(req, res) {
+                res.status(500).send({ message: 'Server error' });
+            },
+        },
+    ]);
+    const client = new Client({ buildServerUri: `http://127.0.0.1:${port}` });
+
+    const result = client.publishAssets('podlet1', ['first.js', 'second.js']);
+
+    await expect(result).rejects.toMatchSnapshot();
```

Comment:
`toThrowErrorMatchingSnapshot()`?

Reply:
becomes quite verbose and turns a 1 line assertion into 3. Also, looking at it, it might actually be nicer to the way it is. Currently:
```
[ValidationError: Invalid 'tag' argument given when attempting to publish assets. "tag" must be a string]
```
would become:
```
Invalid 'tag' argument given when attempting to publish assets. "tag" must be a string
```
Useful to see that it is a ValidationError.